msgspec/_core.c

```diff
@@ -57,11 +58,78 @@ ms_popcount(uint64_t i) {                            \
 #define MS_UNICODE_EQ(a, b) _PyUnicode_EQ(a, b)
 #endif
 
+#if defined(Py_GIL_DISABLED) && !PY314_PLUS
+#error "Py_GIL_DISABLED is only supported in Python 3.14+"
+#endif
+
 #if PY314_PLUS
-#define MS_IMMORTAL_INITIAL_REFCNT _Py_IMMORTAL_INITIAL_REFCNT
+#ifdef Py_GIL_DISABLED
+#define _PyObject_HEAD_INIT(type)    \
```

Comment:
Last comment that might be worth transcribing.

---

You override a private CPython macro with your own initializer. That is brittle across micro-releases and platforms.

Fix: Do not redefine `_PyObject_HEAD_INIT`. Options:

- Keep the existing static sentinel approach but guarded by CPython’s own macros without shadowing names, or
- Initialize sentinels at module init with public API (allocate, set type, optionally mark immortal if a stable API exists; otherwise accept non-immortal + adjusted tests). Given you already relaxed `getrefcount` asserts, the runtime init path is acceptable and far more stable.